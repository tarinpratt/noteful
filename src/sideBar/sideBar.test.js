import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import SideBar from './sideBar'

describe(`SideBar component`, () => {
const props = {
folders: [
  {
    id: "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
    name: "Important"
  },
  {
    id: "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
    name: "Super"
  },
  {
    id: "b07162f0-ffaf-11e8-8eb2-f2801f1b9fd1",
    name: "Spangley"
  }
]
}

it('renders Folders', () => {
  const wrapper = shallow(<SideBar />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders Folders list w/ props', () => {
  const ul = shallow(<SideBar {...props} />)
  expect(toJson(ul)).toMatchSnapshot()
})

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
    <SideBar />
    </BrowserRouter>, 
  div
  )
  ReactDOM.unmountComponentAtNode(div);
})
})