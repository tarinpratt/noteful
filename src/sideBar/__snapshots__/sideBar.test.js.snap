// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component renders Folders 1`] = `
<Fragment>
  <ul
    className="folderList_List"
  />
  <Link
    to="/add-folder"
  >
    <button
      className="addFolder"
    >
      Add Folder
    </button>
  </Link>
</Fragment>
`;

exports[`SideBar component renders Folders list w/ props 1`] = `
<Fragment>
  <ul
    className="folderList_List"
  />
  <Link
    to="/add-folder"
  >
    <button
      className="addFolder"
    >
      Add Folder
    </button>
  </Link>
</Fragment>
`;

exports[`SideBar component renders Folders w/ props 1`] = `
<Fragment>
  <ul
    className="folderList_List"
  />
  <Link
    to="/add-folder"
  >
    <button
      className="addFolder"
    >
      Add Folder
    </button>
  </Link>
</Fragment>
`;
